### PR TITLE
#6 Fibonacci First Attempt

### PR DESCRIPTION
**Pull request by dipakthapaofficial**
Each new term in the Fibonacci sequence is generated by adding the previous two terms. By starting with 1 and 2, the first 10 terms will be:

1, 2, 3, 5, 8, 13, 21, 34, 55, 89, ...

By considering the terms in the Fibonacci sequence whose values do not exceed four million, find the sum of the even-valued terms.

### CODE REVIEW

File: src/com/lftechnology/java/training/dipak/fibonacci/EvenFibonacci.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	private final static Logger LOGGER = Logger.getLogger(EvenFibonacci.class.getName());
+	private int series[];
+	private int index=0;
+	
+	EvenFibonacci(int range){
+		series=new int[range];
+	}
+	/**
+	 * <p>This method generates the fibonacci series for the given range.</p>
+	 * @param  range
```
Comment (utsabneupane) [inline src/com/lftechnology/java/training/dipak/fibonacci/EvenFibonacci.java]: define datatype or class of parameter you are passing

File: src/com/lftechnology/java/training/dipak/fibonacci/EvenFibonacci.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+import java.util.Scanner;
+import java.util.logging.Logger;
+
+/*Each new term in the Fibonacci sequence is generated by adding the previous two terms. By starting with 1 and 2, the first 10 terms will be:
+
+1, 2, 3, 5, 8, 13, 21, 34, 55, 89, ...
+
+By considering the terms in the Fibonacci sequence whose values do not exceed four million, find the sum of the even-valued terms.
+*/
```
Comment (bhuwang) [inline src/com/lftechnology/java/training/dipak/fibonacci/EvenFibonacci.java]: if you use formatter your java doc will be breakdown. 
Format code with Ctrl+Shift+F and you will find what will happen
Use html <pre></pre> tag to format your code properly.
Author is missing in class level java doc
Do not add new line between java doc and method/class declaration.

File: src/com/lftechnology/java/training/dipak/fibonacci/EvenFibonacci.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+			num1=num2;
+			num2=sum;
+		}		
+	}
+	public boolean checkEven(int number){
+		if(number % 2 == 0){
+			return true;
+		}else{
+			return false;
+		}
```
Comment (bhuwang) [inline src/com/lftechnology/java/training/dipak/fibonacci/EvenFibonacci.java:44]: replace if else with

return (number % 2 == 0)

File: src/com/lftechnology/java/training/dipak/fibonacci/EvenFibonacci.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+		for(;;){
+			LOGGER.info("Enter the range upto which the sum even numbers of fibonacci series is to be calculated");
+			Scanner scanner=new Scanner(System.in);
+			try{
+				int range=Math.abs(Integer.parseInt(scanner.nextLine()));
+				EvenFibonacci ef1=new EvenFibonacci(range);
+				ef1.generateSeries(range);
+				int sum=ef1.getEvenNumberSum();				
+				LOGGER.info("The sum of even numbers of fibonacci series whose values do not exceed "+range+" ="+sum);
+				scanner.close();
```
Comment (bhuwang) [inline src/com/lftechnology/java/training/dipak/fibonacci/EvenFibonacci.java]: Again you are closing the resources without taking care of them. Visit my previous suggestions on how to close the open resources and what are the options available.

**Comment by bhuwang** [comment]
Can one of the admins verify this patch?

**Pull request merged**